stetho/src/main/java/com/facebook/stetho/inspector/network/DownloadingAsyncPrettyPrinterFactory.java

```diff
@@ -149,18 +150,31 @@ public PrettyPrinterDisplayType getDisplayType() {
   }
 
   private static class Request implements Callable<String> {
-    private URL url;
+    private final URL url;
 
     public Request(URL url) {
       this.url = url;
     }
 
+    @Nullable
     @Override
-    public String call() throws IOException {
-      InputStream urlStream = url.openStream();
-      String result = Util.readAsUTF8(urlStream);
-      urlStream.close();
-      return result;
+    public String call() throws IOException, IllegalArgumentException {
+      NetworkPeerManager peerManager = NetworkPeerManager.getInstanceOrNull();
+      if (peerManager == null) {
+        throw new IllegalArgumentException("NetworkPeerManager cannot be null");
+      }
+      AsyncPrettyPrinterSchemaManager schemaFileManager
+          = peerManager.getPrettyPrinterSchemaManager();
+      String cacheSchema = schemaFileManager.readCache(url);
```

Comment:
Reading the cache should be done before invoking the `Executor` The reason you need the executor model is because of the fact that you are scheduling the pull from the network on the executor. Now that you are pulling in from an in memory cache there is no need to thread hop.